pkg/_pub_shared/lib/utils/http.dart

```diff
@@ -47,23 +47,41 @@ Future<K> httpGetWithRetry<K>(
   Uri uri, {
   required FutureOr<K> Function(http.Response response) responseFn,
   int maxAttempts = 3,
+
+  /// The HTTP client to use.
+  ///
+  /// Note: when the client is specified, the inner loop will not create a new client object on retries.
+  http.Client? client,
+  Map<String, String>? headers,
+
+  /// Per-request time amount that will be applied on the overall HTTP request.
+  Duration? timeout,
```

Comment:
I think you  (in principle at least) need to put dartdoc documentation of these in the main function dartdoc.

Perhaps rename this to clarify that it is per request.
```suggestion
  Duration? perRequestTimeout,
```